pkg/api/http_server.go

```diff
@@ -113,6 +113,21 @@ func (hs *HTTPServer) AddMiddleware(middleware macaron.Handler) {
 	hs.middlewares = append(hs.middlewares, middleware)
 }
 
+func (hs *HTTPServer) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+	if hs.Cfg.IsLiveEnabled() && strings.HasPrefix(request.URL.Path, "/api/live/push") {
```

Comment:
rather than checking on each request, should we only register this handler if live is enabled

Reply:
this will require different `http.Handler` implementations - which in turn makes code harder to follow.  The overhead should be really negligible since no allocations here. Also maybe `IsLiveEnabled` check will go away soon?